pkg/assets/assets.go

```diff
@@ -102,13 +104,30 @@ func assetFromTemplate(name string, tb []byte, data interface{}) (*Asset, error)
 	return &Asset{Name: name, Data: bs}, nil
 }
 
-type FileInfoPredicate func(os.FileInfo) bool
+type FileInfoPredicate func(string, os.FileInfo) bool
 
 // OnlyYaml is a predicate for LoadFilesRecursively filters out non-yaml files.
-func OnlyYaml(info os.FileInfo) bool {
+func OnlyYaml(_ string, info os.FileInfo) bool {
 	return strings.HasSuffix(info.Name(), ".yaml") || strings.HasSuffix(info.Name(), ".yml")
 }
 
+// InstallerFeatureSet returns a predicate for LoadFilesRecursively that filters manifests
+// based on the specified FeatureSet.
+func InstallerFeatureSet(featureSet string) FileInfoPredicate {
+	targetFeatureSet := "Default"
+	if len(featureSet) > 0 {
+		targetFeatureSet = featureSet
+	}
+	return func(path string, info os.FileInfo) bool {
+		data, err := os.ReadFile(path)
+		if err != nil {
+			panic(err)
+		}
+		manifestFeatureSet := resourceread.ReadUnstructuredOrDie(data).GetAnnotations()["release.openshift.io/feature-set"]
+		return len(manifestFeatureSet) == 0 || manifestFeatureSet == targetFeatureSet
```

Comment:
it's a comma separately list, so split on commas and then check all the values

Reply:
Done.